oshi-core/src/main/java/oshi/software/os/OSThread.java

```diff
@@ -0,0 +1,161 @@
+     */
+
+    String getName();
+
+    /**
+     * <p>
+     * Getter for the field <code>state</code>.
+     * </p>
+     *
+     * @return Returns the execution state of the task/thread.
+     */
+    State getState();
+
+    /**
+     * Gets cumulative CPU usage of this task/thread.
+     * <p>
+     * This calculation sums CPU ticks across all processors.
+     *
+     * @return The proportion of up time that the thread was executing in kernel or
+     *         user mode.
+     */
+    double getCpuLoadCumulative();
+
+    /**
+     * <p>
+     * Getter for the field <code>owningProcessId</code> which is the parent process of this thread.
+     * </p>
+     * @return The owning process of this thread.
+     */
+    int getOwningProcessId();
+
+    /**
+     * <p>
+     * Getter for the field <code>startMemoryAddress</code> which is the memory address above which this thread can run.
+     * </p>
+     *
+     * @return The start address.
+     */
+    long getStartMemoryAddress();
+
+    /**
+     * <p>
+     * Getter for the field <code>contextSwitches</code> which gives a point in time snapshot of the context switches the thread has done.
+     * Since the context switches could be voluntary and non-voluntary, this gives the sum of both.
+     * </p>
+     * @return sum of both voluntary and involuntary context switches.
+     */
+    long getContextSwitches();
+
+    /**
+     * <p>
+     * Getter for the field <code>minorFaults</code>, which gives the number of minor faults the thread has made
+     * which have not required loading a memory page disk.
+     * </p>
+     * @return minor faults.
+     */
+    long getMinorFaults();
+
+    /**
+     * <p>
+     * Getter for the field <code>minorFaults</code>, which gives the number of major faults the thread has made
+     * which have required loading a memory page disk.
+     * </p>
+     * @return minor faults.
+     */
+    long getMajorFaults();
+
+    /**
+     * <p>
+     * Getter for the field <code>kernelTime</code>.
+     * </p>
+     *
+     * @return Returns the number of milliseconds the task/thread has executed in
+     *         kernel/system mode.
+     */
+    long getKernelTime();
+
+    /**
+     * <p>
+     * Getter for the field <code>userTime</code>.
+     * </p>
+     *
+     * @return Returns the number of milliseconds the task/thread has executed in user
+     *         mode.
+     */
+    long getUserTime();
+
+    /**
+     * <p>
+     * Getter for the field <code>upTime</code>.
+     * </p>
+     *
+     * @return Returns the number of milliseconds since the task/thread started.
+     */
+    long getUpTime();
+
+    /**
+     * <p>
+     * Getter for the field <code>startTime</code>.
+     * </p>
+     *
+     * @return Returns the start time of the task/thread in number of milliseconds since
+     *         January 1, 1970.
+     */
+    long getStartTime();
+
+    /**
+     * Attempts to updates process attributes. Returns false if the update fails,
```

Comment:
One question regarding updateAttributes method. Java docs says that it will return false if something, shouldn't we have a try catch block to return false in case of any exception?

Reply:
We don't try to catch exceptions there either, so the program would probably halt on an exception.  We should handle exceptions at the lowest level so these update methods never see them.